Library/Homebrew/formulary.rb

```diff
@@ -122,14 +122,14 @@ def initialize(bottle_name)
       super name, Formulary.path(full_name)
     end
 
-    def get_formula(spec, alias_path: nil)
-      formula = super
+    def get_formula(spec, alias_path: nil) # rubocop:disable Lint/UnusedMethodArgument
+      bottle_version = Utils::Bottles.resolve_version @bottle_filename
+      formula_path = "#{name}/#{bottle_version}/.brew/#{name}.rb"
+      contents = Utils.popen_read "tar", "-xO", "-f", @bottle_filename, "--strip-components=3", formula_path
+      raise BottleFormulaUnavailableError, formula_path unless $CHILD_STATUS.success?
+      formula = Formulary.from_contents name, HOMEBREW_CELLAR/formula_path, contents, spec
+      formula.bottle_specification.sha256 @bottle_filename.sha256 => Utils::Bottles.tag
```

Comment:
What happens without this, out of interest? I'm 👍 on it, just curious.

Reply:
Because the formula stored in the bottle does not have a `bottle` block, the `bottle` method returns `nil`, and causes the error. In principle that error would have also been possible before if the formula in the repo did not contain a bottle block, which is uncommon.
```
==> Pouring the cached bottle
Error: undefined method `compatible_cellar?' for nil:NilClass
/Users/sjackman/.homebrew/Library/Homebrew/formula_installer.rb:104:in `pour_bottle?'
```
I'll add a commit to address this issue.

Reply:
Done.